syft/pkg/cataloger/java/internal/maven/resolver.go

```diff
@@ -293,7 +335,7 @@ func (r *mavenResolver) findPom(ctx context.Context, groupID, artifactID, versio
 
 	// resolve via network maven repository
 	if pom == nil && r.cfg.UseNetwork {
```

Comment:
I think this was always a problem -- but this is checking the local var `pom` which [isn't being overridden in  by the local repo search](https://github.com/anchore/syft/pull/3363/files#diff-005fd8c5b8db8fa24b7ca155ab6e0b9f103f6e7f88bdb1bf459623dd8900cea1R328), it's being shadowed. So the [first check](https://github.com/anchore/syft/pull/3363/files#diff-005fd8c5b8db8fa24b7ca155ab6e0b9f103f6e7f88bdb1bf459623dd8900cea1R320) will always take precedence.

Reply:
I see that `pom` is always going to be nil here, but the logic is always supposed to run in this order to minimize network usage: check in-memory cache, check local repository, check remote. Is there anything to do other than just remove the `pom == nil` condition?

Reply:
I think removing the condition is safe since there is the early return for a hit when searching m2 anyway.